src/components/RadioWithTextField.tsx

```diff
@@ -19,27 +19,34 @@ type Props = {
   title: React.ReactNode;
   label: string;
   field:
-    | 'isEstabilishedRegulatoryProvision'
-    | 'fromBelongsRegulatedMarket'
-    | 'isFromIPA'
-    | 'isConcessionaireOfPublicService';
+  | 'isEstabilishedRegulatoryProvision'
+  | 'fromBelongsRegulatedMarket'
+  | 'isFromIPA'
+  | 'isConcessionaireOfPublicService';
   errorText: string;
   onRadioChange: (field: any, value: any) => void;
   onTextFieldChange: (open: boolean, field: string, value: string) => void;
+  isIPA?: boolean;
+  ipaCode?: string;
 };
 
+// eslint-disable-next-line sonarjs/cognitive-complexity
 export function RadioWithTextField({
   title,
   label,
   field,
   errorText,
   onRadioChange,
   onTextFieldChange,
+  isIPA,
+  ipaCode
 }: Props) {
   const theme = useTheme();
   const { t } = useTranslation();
 
   const [openTextField, setOpenTextField] = useState<boolean>(false);
+  const [checked, setChecked] = useState<boolean | null>(null);
```

Comment:
why null? use boolean as type and set false as default value

Reply:
null because we need the value of the state variable as to be null in order to let the radios that doesn't have the logic of IPA without any type value setted or initialized.